### ISSUE TITLE
Session ID does not reset for web analytic 

### ISSUE THREAD

**Issue by TeddyCr** [open-metadata/OpenMetadata#8540]
We notice 2 things with the sessionId in the web analytic event that is sent:
1. the same sessionId is passed for all users -> to reproduce create different users and check the session ID passed for each user in the request.
2. the session ID does not expire. what we expect -> if a user is not active for 30 minutes then the session ID should reset. Example: a user comes at 12:00 (session ID set to 1234) visits different pages, then leaves the website. Comes back at 12:15 (session ID is still 1234) does other things on the platform leaves again and comes back at 1:00PM (new session ID should be set 56789).

Currently a new session ID is set only if different browsers are being used.

**Comment by harshach** [comment]
3. After close of the browser page before 30 mins expired should we reset as another session?

**Comment by TeddyCr** [comment]
I think it should be the same session. I basically went with Google Analytics definition of a session

**Comment by Sachin-chaurasiya** [comment]
> I think it should be the same session. I basically went with Google Analytics' definition of a session

@TeddyCr @harshach, Here are the case for time-based expiration.
- Default we will set the expiry of 30 minutes
- If the user is inactive for 30 minutes then the session will expire and a new session will be created.
- If the user is active and interacting with pages then we will reset the expiration by extending the expiry with additional 30 minutes.

Example : 
Let's say the user visited the home page and the session starts (30 minutes), now the user is on the same page for 29 minutes and then he started interacting by visiting another page then we will extend the expiry that is

Time left = 30 - 29 ( time spent by user ) = 1 minute
New Expiry = time left + 30 minutes = 31 minutes

**Comment by TeddyCr** [comment]
@Sachin-chaurasiya it should be always 30 minutes from last interaction and not 30 + time left. In you example above it would be 30 minutes and not 31 minutes otherwise we'll have scenarios such as user starts session (30 minutes) after 1 minute moves to another page (noew time left would be 29+30 = 59) moves to another page after 30 seconds (time left in session is now 58.5 + 30 = 88.5), etc.

Basically expiration is last timestamp interaction + 30 minutes.

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya it should be always 30 minutes from last interaction and not 30 + time left. In you example above it would be 30 minutes and not 31 minutes otherwise we'll have scenarios such as user starts session (30 minutes) after 1 minute moves to another page (noew time left would be 29+30 = 59) moves to another page after 30 seconds (time left in session is now 58.5 + 30 = 88.5), etc.
> 
> Basically expiration is last timestamp user intercated + 30 minutes.

Thanks, @TeddyCr, Got it.

**Comment by Sachin-chaurasiya** [comment]
> > @Sachin-chaurasiya it should be always 30 minutes from last interaction and not 30 + time left. In you example above it would be 30 minutes and not 31 minutes otherwise we'll have scenarios such as user starts session (30 minutes) after 1 minute moves to another page (noew time left would be 29+30 = 59) moves to another page after 30 seconds (time left in session is now 58.5 + 30 = 88.5), etc.
> > Basically expiration is last timestamp user intercated + 30 minutes.
> 
> Thanks, @TeddyCr, Got it.

@TeddyCr , Can you verify this example?

Let's say the expiry is at "5:45:23 PM", and the user spent some time and then Interact with other pages in 2 minutes so the expiry time will be extended to "5:47:23 PM".
